includes/connect/class-wc-stripe-connect.php

```diff
@@ -0,0 +1,185 @@
+				return $response;
+			}
+
+			return $this->save_stripe_keys( $response );
+		}
+
+		/**
+		 * Handle redirect back from oauth-init or credentials reset
+		 */
+		public function maybe_handle_redirect() {
+			if ( ! is_admin() ) {
+				return;
+			}
+
+			// redirect from oauth-init
+			if ( isset( $_GET['wcs_stripe_code'], $_GET['wcs_stripe_state'] ) ) {
+
+				$response = $this->connect_oauth( $_GET['wcs_stripe_state'], $_GET['wcs_stripe_code'] );
+				wp_safe_redirect( remove_query_arg( array( 'wcs_stripe_state', 'wcs_stripe_code' ) ) );
+				exit;
+			// redirect from credentials reset
+			} elseif ( isset( $_GET['reset_stripe_api_credentials'] ) ) {
+
+				$this->clear_stripe_keys();
+				wp_safe_redirect( remove_query_arg( array( 'reset_stripe_api_credentials' ) ) );
+				exit;
+			}
+
+		}
+
+		/**
+		 * Saves stripe keys after OAuth response
+		 *
+		 * @param  array $result OAuth response result.
+		 *
+		 * @return array|WP_Error
+		 */
+		private function save_stripe_keys( $result ) {
+
+			if ( ! isset( $result->publishableKey, $result->secretKey ) ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				return new WP_Error( 'Invalid credentials received from WooCommerce Connect server' );
+			}
+
+			$is_test                                = false !== strpos( $result->publishableKey, '_test_' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$prefix                                 = $is_test ? 'test_' : '';
+			$default_options                        = $this->get_default_stripe_config();
+			$options                                = array_merge( $default_options, get_option( self::SETTINGS_OPTION, array() ) );
+			$options['enabled']                     = 'yes';
+			$options['testmode']                    = $is_test ? 'yes' : 'no';
+			$options[ $prefix . 'publishable_key' ] = $result->publishableKey; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$options[ $prefix . 'secret_key' ]      = $result->secretKey; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+			// While we are at it, let's also clear the account_id and
+			// test_account_id if present.
+			unset( $options['account_id'] );
+			unset( $options['test_account_id'] );
+
+			update_option( self::SETTINGS_OPTION, $options );
+
+			return $result;
+		}
+
+		/**
+		 * Clears keys for test or production (whichever is presently enabled).
+		 */
+		private function clear_stripe_keys() {
+
+			$options = get_option( self::SETTINGS_OPTION, array() );
+
+			if ( 'yes' === $options['testmode'] ) {
+				$options['test_publishable_key'] = '';
+				$options['test_secret_key']      = '';
+			} else {
+				$options['publishable_key'] = '';
+				$options['secret_key']      = '';
+			}
+
+			// While we are at it, let's also clear the account_id and
+			// test_account_id if present.
+			unset( $options['account_id'] );
```

Comment:
Should these unsets be within the conditionals? To avoid blowing away both account IDs? i.e. unset test_account_id if in testmode, unset the live account_id if not.

Reply:
Truthfully, I'm not certain. It seems to me that we are wanting both of these values to be removed if they are present, but to be safe, I've moved the unsets to within the above conditional logic in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267/commits/4c66a0801c9d8bbcb645bb3aec3646eecfd92e25